src/azul/drs.py

```diff
@@ -97,10 +155,18 @@ def _uri_to_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
         compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
         are not.
         """
-        parsed = furl(drs_uri)
-        scheme = 'drs'
-        require(parsed.scheme == scheme,
-                f'The URI {drs_uri!r} does not have the {scheme!r} scheme')
+        try:
+            parsed = furl(drs_uri)
+        except ValueError as e:
+            require(e.args[0].startswith('Invalid port'))
```

Comment:
```suggestion
            # So-called DRS "URIs" (https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.3.0/docs/) for Compact Identifiers (https://www.nature.com/articles/sdata201829) are NOT URIs according to RFC 3986 (https://datatracker.ietf.org/doc/html/rfc3986) so we can't use off-the-shelf URI parsers.
            require(e.args[0].startswith('Invalid port'))
```

Reply:
You also need to be more defensive when determining if the ValueError instance matches your expectations. It might not have any args, for example. Any ValueError that doesn't match should not cause secondary exceptions like IndexError, but should be re-raised.